### DISCUSSION TITLE
pmd ARIMA captures spikes better than statsforecast arima

### DISCUSSION THREAD

**Discussion by laniakea** [Nixtla/statsforecast#944] [Q&A] [answered]
Hi there,
I'm using SSA (singular spectrum analysis) with both pmd ARIMA and statsforecast arima. SSA creates a series of time series that do not have timeindex, although they are correlated.
I predict each series using ARIMA and combine the results as an overall prediction.
The issue is I get better prediction results from pmd ARIMA, when both have stepwise=False and the rest of the parameters are the same.
For statsforecast I set the freq=1, as SSA extracts the seasonality and each series have a different seasonality.

My guess is AutoARIMA expects a specific parameter to be set (like seasonality or something similar in pandas?) or the way statsforecast optimizes ARIMA parameters is not doing as good as pmd ARIMA.

Your feedback and help is really appreciate it.

**Comment by jmoralez** [comment]
Can you provide an example?

**Comment by laniakea** [comment]
Sorry, I've worked on the issue since then and realized that when I set seasonality to 7 (there is weekly pattern) it captures the spikes.
 Since I've set the stepwise=False, with seasonality 7 it perform extremely slow.

I have series of length 30 (small number of observations per series). Each are decomposed using SSA and get predicted individually. SSA decomposes each time series by eigen decomposition of the covariance matrix. The largest eigencomponent represents the trend and the smaller components will have quasi-periodic shapes and the smallest are simply noise. I'm not sure why ARIMA is expecting the seasonality parameter? Is seasonality used to define the lags search space? Not sure why it's needed as with SSA the seasonality is not necessary unless it's used for lags. I use this to predict each component:

`model = StatsForecast(models=[AutoARIMA(
                                            stepwise=False, 
                                            max_p=7, 
                                            max_q=1,
                                            max_d= 0,
                                            seasonal=True, 
                                            season_length=7)],
                                            n_jobs=-1, freq=1)`
                                            
Each time I forecast one component produced by SSA and then predict the component K steps. If I set stepwise=True, the prediction error becomes unacceptably large.

There are only 30 observations per each series, and it is extremely slow. I'm not sure what I am missing here?

**Reply by jmoralez** [reply]
If you could provide an example along with your expectations it'd help a lot. If you set a seasonality then the AutoARIMA tries several SARIMA models (which is why it needs a season_length) which can take some time, so if you don't want it to be "extremely slow" you can remove the seasonality component or use a fixed ARIMA model.

**Reply by laniakea** [reply]
Thank you. Assuming the process is stationary with seasonality, SSA should be able to extract the components and each component can be forecasted by a simple AR model providing proper lags, e.g. Autoreg(data, lags=[1,2,5,6,9]).
In an ideal scenario this will fit the series perfectly.
Is it possible to specify the lags for the AR part of you ARIMA model then? In statsmodels or pmd arima, it is possible to pass a list as the p and q. e.g. ARIMA(data, p=[1,3,4,5,8], q=[1,2,5], ...)

**Reply by jmoralez** [reply]
You can only set the total number, e.g.
```python
from statsforecast.models import ARIMA

model = ARIMA(order=(8, 0, 5))  # (p, d, q)
```

**Reply by laniakea** [reply]
Thanks. When we set AR order to 8, is it going to consider only lag 8th or all the lags from 1 to 8 or it selects the lags (e.g. 1,3,5,8) based on some information measure? Is it possible to chose the method?

**Reply by jmoralez** [reply, marked as answer]
It will use 1 to 8. For the methods you can choose `ML` or  `CSS-ML`

**Reply by laniakea** [reply]
Thanks so much
